### PR TITLE
Use python-gflags instead of argparse

### PR DESCRIPTION
**Pull request by funrollloops**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -427,52 +430,43 @@ def StartServer(port, syncer):
...
-  parser.add_argument('--frontend',
-                      default='http://nebree8.appspot.com',
-                      help='Frontend server to sync with, if any')
-  parser.add_argument('--fe_poll_freq',
-                      type=int,
-                      default=5,
-                      help='Frontend polling frequency in seconds')
-  parser.set_defaults(fake=False, logtostderr=True)
-  args = parser.parse_args()
+  import sys
```
Comment (audiodude) [inline server.py]: I would put the sys import at the top, as per usual.

File: server.py [hunk 2]
```diff
@@ -427,52 +430,43 @@ def StartServer(port, syncer):
...
+  FLAGS(sys.argv)
 
   # Set up logging
   rootLogger = logging.getLogger()
-  rootLogger.setLevel(getattr(logging, args.loglevel.upper()))
-  if args.logfile:
-    rootLogger.addHandler(logging.FileHandler(args.logfile))
-  if args.logtostderr:
+  rootLogger.setLevel(getattr(logging, FLAGS.loglevel.upper()))
```
Comment (audiodude) [inline server.py]: The old implementation had a default value for loglevel. Where are you setting the default for FLAGS.loglevel? Same for other flags that used to have defaults.

File: server.py [hunk 3]
```diff
@@ -427,52 +430,43 @@ def StartServer(port, syncer):
...
-                      default='info',
-                      help='Log level (debug, info, warning, error)')
-  parser.add_argument('--frontend',
-                      default='http://nebree8.appspot.com',
-                      help='Frontend server to sync with, if any')
-  parser.add_argument('--fe_poll_freq',
-                      type=int,
-                      default=5,
-                      help='Frontend polling frequency in seconds')
-  parser.set_defaults(fake=False, logtostderr=True)
```
Comment (audiodude) [inline server.py]: What is the equivalent of this line?

File: setup.sh [hunk 4]
```diff
@@ -13,10 +13,10 @@ function install_once {
     python-smbus python-pip i2c-tools \
     libxml2 python-dev python-pycurl python-pyquery \
     imagemagick
-  run sudo pip install --upgrade pip
-  run sudo pip install --upgrade \
+  run sudo -H pip install --upgrade pip
```
Comment (audiodude) [inline setup.sh]: I looked up `sudo -H` but I don't understand why you want to use root's home directory instead of the current user's?

Comment (funrollloops) [inline setup.sh]: pip and/or npm has been complaining to me that the .cache directory is owned by the wrong user.

**Comment by audiodude** [comment]
LGTM

**Pull request merged**